src/Map/TileMap.cs

```diff
@@ -40,9 +40,12 @@ class TileMap
 
         private MouseProcessor mouseProcessor = new MouseProcessor();
 
-        // TODO: Replace with actual object representation, rather than just generic object
+        //Dictionary containing all entities and their positions on the map
         Dictionary<Entity, Vector2> PlacedObjects = new Dictionary<Entity, Vector2>();
 
+        //DIctionary containing all entities and the team that they belong to
+        Dictionary<Entity, Team> TeamDictionary = new Dictionary<Entity, Team>();
```

Comment:
Yea, it seems odd to have Team assignment here. Maybe we should rethink where we store PlacedObjects. As long as the map has a reference to the Entities and where they are placed, it doesn't need to be saved here. Maybe we start storing location on entity, and have a singleton list of teams. Then the teams keep the reference to the entities on them.